src/thor/dijkstras.cc

```diff
@@ -238,23 +238,27 @@ void Dijkstras::ExpandInner(baldr::GraphReader& graphreader,
 
     // Only needed if you want to connect with a reverse path - for reverse mode,
     // these were populated earlier
-    if (expansion_direction == ExpansionType::forward) {
+    if (FORWARD) {
       t2 = tile;
       oppedgeid = graphreader.GetOpposingEdgeId(edgeid, t2);
     }
 
     // Add edge label, add to the adjacency list and set edge status
     uint32_t idx = bdedgelabels_.size();
     *es = {EdgeSet::kTemporary, idx};
+    auto destonly =
+        FORWARD ? directededge->destonly() || (costing_->is_hgv() && directededge->destonly_hgv())
+                : opp_edge->destonly() || (costing_->is_hgv() && opp_edge->destonly_hgv());
```

Comment:
lets think about the directionality of edgelabel properties

Reply:
So I looked at the code a bit again: on reverse search we add the edge IDs of the expanded edges (i.e. the opposing ones of the final path), but we do add the _cost_ of their opposing edge which also sets the sortcost for the adjacency queue. While we don't have directionality of destonly _yet_ I still think this is the right way to go. See #4456 where IMO we're not adhering to this principle either.

Reply:
I refactored this a bit. Now it distinguishes between forward and reverse when adding labels during expansion.